nifi-commons/nifi-record-path/src/test/java/org/apache/nifi/record/path/TestRecordPath.java

```diff
@@ -1837,7 +1838,41 @@ public void testUnescapeJson() {
                     put("firstName", "John");
                     put("age", 30);
                 }}, "json_str"),
-                RecordPath.compile("unescapeJson(/json_str, 'true')").evaluate(recordFromMap).getSelectedFields().findFirst().orElseThrow(IllegalStateException::new).getValue()
+                RecordPath.compile("unescapeJson(/json_str, 'true')").evaluate(recordFromMap).getSelectedFields().findFirst().orElseThrow(AssertionError::new).getValue()
+        );
+
+        // test nested Record converted from Map Object
+        final Record nestedRecordFromMap = new MapRecord(schema,
+                Collections.singletonMap(
+                        "json_str",
+                        "{\"firstName\":\"John\",\"age\":30,\"addresses\":[{\"address_1\":\"123 Fake Street\"}]}")
```

Comment:
Would probably be cleaner here to use a text block here:
```
"""
{"firstName":"John","age":30,"address"...}"""
```

Reply:
I'd been sticking with Java 8 compatible code to get this onto the `support/1.x` branch, but if we're happy with it being only NiFi 2.x now, I'll update these tests